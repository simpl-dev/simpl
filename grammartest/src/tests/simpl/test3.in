grammar ee.cyber.simplicitas.grammartest.expr.Expr;

Id: id==Id;
Id2: id=Id;
Id3: id=;
