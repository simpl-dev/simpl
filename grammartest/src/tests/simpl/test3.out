0:85:GrammarDef {
  header=0:49:PackageDef {
    dottedId=8:49:DottedId {
      head=8:9:Id {
        text="ee"
      }
      tail=List {
        11:15:Id {
          text="cyber"
        }
        17:27:Id {
          text="simplicitas"
        }
        29:39:Id {
          text="grammartest"
        }
        41:44:Id {
          text="expr"
        }
        46:49:Id {
          text="Expr"
        }
      }
    }
  }
  opts=(null)
  rules=List {
    53:63:NonTerminalDef {
      alt=57:62:AltList {
        lst=57:62:MatchList {
          m=List {
            57:59:Match {
              modifier=(null)
              name=57:58:Id {
                text="id"
              }
              token=(null)
            }
            61:62:Match {
              modifier=(null)
              name=(null)
              token=61:62:MatchToken {
                alt=(null)
                ref=61:62:RuleRef {
                  id=61:62:Id {
                    text="Id"
                  }
                }
                str=(null)
              }
            }
          }
        }
        lst2=List {
        }
      }
      body=(null)
      name=53:54:Id {
        text="Id"
      }
    }
    65:75:NonTerminalDef {
      alt=70:74:AltList {
        lst=70:74:MatchList {
          m=List {
            70:74:Match {
              modifier=(null)
              name=70:71:Id {
                text="id"
              }
              token=73:74:MatchToken {
                alt=(null)
                ref=73:74:RuleRef {
                  id=73:74:Id {
                    text="Id"
                  }
                }
                str=(null)
              }
            }
          }
        }
        lst2=List {
        }
      }
      body=(null)
      name=65:67:Id {
        text="Id2"
      }
    }
    77:85:NonTerminalDef {
      alt=82:84:AltList {
        lst=82:84:MatchList {
          m=List {
            82:84:Match {
              modifier=(null)
              name=82:83:Id {
                text="id"
              }
              token=(null)
            }
          }
        }
        lst2=List {
        }
      }
      body=(null)
      name=77:79:Id {
        text="Id3"
      }
    }
  }
}
